pkg/kubelet/volumes.go

```diff
@@ -263,23 +251,27 @@ func (kl *Kubelet) getPodVolumesFromDisk() map[string]cleanerTuple {
 	return currentVolumes
 }
 
+// newVolumeMounterFromPlugins attempts to find a plugin by volume spec and then create a physicalMounter from
+// the volume spec, pod and volume options for that volume.
+// This function will return an error if plugin was not found or could not mount the volume
+// otherwise it returns the phyiscalMounter
 func (kl *Kubelet) newVolumeMounterFromPlugins(spec *volume.Spec, pod *api.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	plugin, err := kl.volumePluginMgr.FindPluginBySpec(spec)
 	if err != nil {
 		return nil, fmt.Errorf("can't use volume plugins for %s: %v", spec.Name(), err)
 	}
-	if plugin == nil {
-		// Not found but not an error
-		return nil, nil
-	}
 	physicalMounter, err := plugin.NewMounter(spec, pod, opts)
 	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate volume physicalMounter for %s: %v", spec.Name(), err)
+		return nil, fmt.Errorf("failed to instantiate physicalMounter for volume: %s using plugin: %s with a root cause: %v", spec.Name(), plugin.Name(), err)
 	}
 	glog.V(10).Infof("Used volume plugin %q to mount %s", plugin.Name(), spec.Name())
 	return physicalMounter, nil
 }
 
+// newVolumeAttacherFromPlugins attempts to find a plugin by a persistent volume spec and then create an attacher.
+// This is used when a volume needs to attach to a node before mounting.
+// This func will return an error if the attacher plugin was not found or if the Attacher failed to instantiate.
```

Comment:
The information in this line and below is appropriate content, the tense is a little off.  Use present tense, viz: 'returns an error '